### PR TITLE
:broom: Align dep vers with Spack stable, drop Py3.7

### LINKED ISSUE #2147: Track dependency versions based on Spack stable branch.
We could have our dependencies track the Spack stable branch. Since Spack's a major way Arbor is rolled out (CSCS, JSC, Ebrains), it has an obvious benefit.

### PR DESCRIPTION
**Pull request by brenthuisman**
- Git submodules now align with the highest version available in Spack stable.
- Dropped Py3.7 support.
- Update docs for Sphinx 6+

Fixes #2147

### CODE REVIEW

File: .github/workflows/test-docs.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-20.04]
-        python-version: [3.7]
+        python-version: [3.8]
```
Comment (thorstenhater) [inline .github/workflows/test-docs.yaml:18]: 🎉

File: doc/concepts/decor.rst [hunk 2]
```diff
@@ -176,7 +176,7 @@ Take for example the built-in mechanism for passive leaky dynamics:
 .. _cablecell-scaled-mechs:
 
 4. Scaling Mechanism and Membrane Parameters
-~~~~~~~~~~~~~~~~~~~~~
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (thorstenhater) [inline doc/concepts/decor.rst:179]: Jupp, caught that, too ;) See #2212

File: spack/package.py [hunk 3]
```diff
@@ -93,18 +95,20 @@ class Arbor(CMakePackage, CudaPackage):
     depends_on("py-mpi4py", when="+mpi+python", type=("build", "run"))
 
     # python (bindings)
-    extends("python", when="+python")
-    depends_on("python@3.7:", when="+python", type=("build", "run"))
-    depends_on("py-numpy", when="+python", type=("build", "run"))
     with when("+python"):
+        extends("python")
+        depends_on("python@3.7:", type=("build", "run"))
```
Comment (thorstenhater) [inline spack/package.py:100]: This is still 3.7, why?

File: spack/package.py [hunk 4]
```diff
@@ -93,18 +95,20 @@ class Arbor(CMakePackage, CudaPackage):
...
+        depends_on("python@3.8:", when="@0.9.1:", type=("build", "run"))
+        depends_on("py-numpy", type=("build", "run"))
         depends_on("py-pybind11@2.6:", type="build")
```
Comment (thorstenhater) [inline spack/package.py:103]: Uh. This version is all over the place,
- 0.9.1: 2.6 which is quite old
- 0.7.1: 2.10 
- 0.5.3: 2.8

or am I misunderstand something.

**Review by thorstenhater — CHANGES_REQUESTED** [review]

File: spack/package.py (hunk 3 above)
Comment (brenthuisman) [reply spack/package.py:100]: This is pre-Arbor 0.9.1, so any version released up to now.

File: spack/package.py (hunk 4 above)
Comment (thorstenhater) [reply spack/package.py:103]: Ok, that makes it way less convoluted!

**Review by thorstenhater — APPROVED** [review]

**Pull request merged**